pkg/controllers/termination/controller.go

```diff
@@ -37,25 +39,52 @@ import (
 
 // Controller for the resource
 type Controller struct {
-	terminator    *Terminator
-	cloudProvider cloudprovider.CloudProvider
-	kubeClient    client.Client
+	Terminator *Terminator
+	KubeClient client.Client
+}
+
+// For returns the resource this controller is for.
+func (c *Controller) For() client.Object {
+	return &v1.Node{}
+}
+
+func (c *Controller) Name() string {
+	return "terminator"
 }
 
 // NewController constructs a controller instance
 func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface, cloudProvider cloudprovider.CloudProvider) *Controller {
 	return &Controller{
-		terminator:    &Terminator{kubeClient: kubeClient, cloudProvider: cloudProvider, coreV1Client: coreV1Client},
-		cloudProvider: cloudProvider,
-		kubeClient:    kubeClient,
+		KubeClient: kubeClient,
+		Terminator: &Terminator{kubeClient: kubeClient, cloudProvider: cloudProvider, coreV1Client: coreV1Client,
+			evictionQueue: EvictionQueue{
+				queue:        workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(100*time.Millisecond, 10*time.Second)),
+				coreV1Client: coreV1Client,
+				enqueued:     set.NewSet(),
+				once:         sync.Once{},
+			}},
+	}
+}
+
+func NewControllerWithCustomQueue(kubeClient client.Client, coreV1Client corev1.CoreV1Interface,
```

Comment:
I understand where you're coming from. I created this function precisely for the reason you're talking about, but also because I didn't want to make the EvictionQueue struct variables public.

Instead, I decided to make a terminator constructor. LMK what you think.